// tailwindcss-classnames/lib/index.js
"use strict";
/* eslint-disable */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.invert = exports.hueRotate = exports.grayscale = exports.dropShadow = exports.contrast = exports.brightness = exports.blur = exports.backdropFilter = exports.filter = exports.animation = exports.transitionDelay = exports.transitionTimingFunction = exports.transitionDuration = exports.transitionProperty = exports.opacity = exports.boxShadowColor = exports.boxShadow = exports.backgroundBlendMode = exports.mixBlendMode = exports.tableLayout = exports.borderCollapse = exports.outlineColor = exports.outlineWidth = exports.outlineOffset = exports.ringOffsetWidth = exports.ringOffsetColor = exports.ringOpacity = exports.ringWidth = exports.ringColor = exports.divideWidth = exports.divideOpacity = exports.divideColor = exports.borderWidth = exports.borderRadius = exports.borderOpacity = exports.borderColor = exports.outlineStyle = exports.divideStyle = exports.borderStyle = exports.gradientColorStops = exports.backgroundImage = exports.backgroundSize = exports.backgroundPosition = exports.backgroundColor = exports.backgroundOpacity = exports.backgroundRepeat = exports.backgroundOrigin = exports.backgroundClip = exports.backgroundAttachment = exports.screenReaders = void 0;
exports.cursor = exports.touchAction = exports.scrollBehavior = exports.scrollSnap = exports.userSelect = exports.resize = exports.pointerEvents = exports.appearance = exports.space = exports.margin = exports.padding = exports.gap = exports.gridRowEnd = exports.gridRowStart = exports.gridRow = exports.gridAutoRows = exports.gridTemplateRows = exports.gridColumnEnd = exports.gridColumnStart = exports.gridColumn = exports.gridAutoColumns = exports.gridTemplateColumns = exports.gridAutoFlow = exports.order = exports.flexShrink = exports.flexGrow = exports.flexBasis = exports.flex = exports.justifySelf = exports.justifyItems = exports.justifyContent = exports.placeSelf = exports.placeItems = exports.placeContent = exports.alignSelf = exports.alignContent = exports.alignItems = exports.flexWrap = exports.flexDirection = exports.backdropSepia = exports.backdropSaturate = exports.backdropOpacity = exports.backdropInvert = exports.backdropHueRotate = exports.backdropGrayscale = exports.backdropContrast = exports.backdropBrightness = exports.backdropBlur = exports.sepia = exports.saturate = void 0;
exports.whitespace = exports.verticalAlign = exports.textOverflow = exports.textTransform = exports.textDecoration = exports.textAlign = exports.listStylePosition = exports.fontVariantNumeric = exports.fontStyle = exports.fontSmoothing = exports.transformOrigin = exports.skew = exports.translate = exports.rotate = exports.scale = exports.hardwareAcceleration = exports.strokeWidth = exports.stroke = exports.fill = exports.maxHeight = exports.minHeight = exports.height = exports.maxWidth = exports.minWidth = exports.width = exports.columns = exports.aspectRatio = exports.zIndex = exports.inset = exports.objectPosition = exports.breakAfter = exports.breakInside = exports.breakBefore = exports.visibility = exports.position = exports.overscrollBehavior = exports.overflow = exports.objectFit = exports.isolation = exports.clear = exports.float = exports.container = exports.boxSizing = exports.boxDecorationBreak = exports.display = exports.scrollMargin = exports.scrollPadding = exports.accentColor = exports.willChange = exports.caretColor = void 0;
exports.classnames = exports.TW = exports.textUnderlineOffset = exports.textDecorationThickness = exports.textDecorationColor = exports.textIndent = exports.content = exports.textOpacity = exports.textColor = exports.placeholderOpacity = exports.placeholderColor = exports.listStyleType = exports.lineHeight = exports.letterSpacing = exports.fontWeight = exports.fontSize = exports.fontFamily = exports.textDecorationStyle = exports.wordBreak = void 0;
var tslib_1 = require("tslib");
//////////////////////////////////////////////////////////////////////////////
/// Autogenerated by tailwindcss-classnames CLI. https://git.io/JtsPU
/// DO NOT EDIT THIS FILE DIRECTLY!
//////////////////////////////////////////////////////////////////////////////
var clsx_1 = tslib_1.__importDefault(require("clsx"));
//////////// Accessibility Utility functions
exports.screenReaders = clsx_1.default;
//////////// Backgrounds Utility functions
exports.backgroundAttachment = clsx_1.default;
exports.backgroundClip = clsx_1.default;
exports.backgroundOrigin = clsx_1.default;
exports.backgroundRepeat = clsx_1.default;
exports.backgroundOpacity = clsx_1.default;
exports.backgroundColor = clsx_1.default;
exports.backgroundPosition = clsx_1.default;
exports.backgroundSize = clsx_1.default;
exports.backgroundImage = clsx_1.default;
exports.gradientColorStops = clsx_1.default;
//////////// Borders Utility functions
exports.borderStyle = clsx_1.default;
exports.divideStyle = clsx_1.default;
exports.outlineStyle = clsx_1.default;
exports.borderColor = clsx_1.default;
exports.borderOpacity = clsx_1.default;
exports.borderRadius = clsx_1.default;
exports.borderWidth = clsx_1.default;
exports.divideColor = clsx_1.default;
exports.divideOpacity = clsx_1.default;
exports.divideWidth = clsx_1.default;
exports.ringColor = clsx_1.default;
exports.ringWidth = clsx_1.default;
exports.ringOpacity = clsx_1.default;
exports.ringOffsetColor = clsx_1.default;
exports.ringOffsetWidth = clsx_1.default;
exports.outlineOffset = clsx_1.default;
exports.outlineWidth = clsx_1.default;
exports.outlineColor = clsx_1.default;
//////////// Tables Utility functions
exports.borderCollapse = clsx_1.default;
exports.tableLayout = clsx_1.default;
//////////// Effects Utility functions
exports.mixBlendMode = clsx_1.default;
exports.backgroundBlendMode = clsx_1.default;
exports.boxShadow = clsx_1.default;
exports.boxShadowColor = clsx_1.default;
exports.opacity = clsx_1.default;
//////////// TransitionsAndAnimations Utility functions
exports.transitionProperty = clsx_1.default;
exports.transitionDuration = clsx_1.default;
exports.transitionTimingFunction = clsx_1.default;
exports.transitionDelay = clsx_1.default;
exports.animation = clsx_1.default;
//////////// Filters Utility functions
exports.filter = clsx_1.default;
exports.backdropFilter = clsx_1.default;
exports.blur = clsx_1.default;
exports.brightness = clsx_1.default;
exports.contrast = clsx_1.default;
exports.dropShadow = clsx_1.default;
exports.grayscale = clsx_1.default;
exports.hueRotate = clsx_1.default;
exports.invert = clsx_1.default;
exports.saturate = clsx_1.default;
exports.sepia = clsx_1.default;
exports.backdropBlur = clsx_1.default;
exports.backdropBrightness = clsx_1.default;
exports.backdropContrast = clsx_1.default;
exports.backdropGrayscale = clsx_1.default;
exports.backdropHueRotate = clsx_1.default;
exports.backdropInvert = clsx_1.default;
exports.backdropOpacity = clsx_1.default;
exports.backdropSaturate = clsx_1.default;
exports.backdropSepia = clsx_1.default;
//////////// FlexBox Utility functions
exports.flexDirection = clsx_1.default;
exports.flexWrap = clsx_1.default;
exports.alignItems = clsx_1.default;
exports.alignContent = clsx_1.default;
exports.alignSelf = clsx_1.default;
exports.placeContent = clsx_1.default;
exports.placeItems = clsx_1.default;
exports.placeSelf = clsx_1.default;
exports.justifyContent = clsx_1.default;
exports.justifyItems = clsx_1.default;
exports.justifySelf = clsx_1.default;
exports.flex = clsx_1.default;
exports.flexBasis = clsx_1.default;
exports.flexGrow = clsx_1.default;
exports.flexShrink = clsx_1.default;
exports.order = clsx_1.default;
//////////// Grid Utility functions
exports.gridAutoFlow = clsx_1.default;
exports.gridTemplateColumns = clsx_1.default;
exports.gridAutoColumns = clsx_1.default;
exports.gridColumn = clsx_1.default;
exports.gridColumnStart = clsx_1.default;
exports.gridColumnEnd = clsx_1.default;
exports.gridTemplateRows = clsx_1.default;
exports.gridAutoRows = clsx_1.default;
exports.gridRow = clsx_1.default;
exports.gridRowStart = clsx_1.default;
exports.gridRowEnd = clsx_1.default;
exports.gap = clsx_1.default;
//////////// Spacing Utility functions
exports.padding = clsx_1.default;
exports.margin = clsx_1.default;
exports.space = clsx_1.default;
//////////// Interactivity Utility functions
exports.appearance = clsx_1.default;
exports.pointerEvents = clsx_1.default;
exports.resize = clsx_1.default;
exports.userSelect = clsx_1.default;
exports.scrollSnap = clsx_1.default;
exports.scrollBehavior = clsx_1.default;
exports.touchAction = clsx_1.default;
exports.cursor = clsx_1.default;
exports.caretColor = clsx_1.default;
exports.willChange = clsx_1.default;
exports.accentColor = clsx_1.default;
exports.scrollPadding = clsx_1.default;
exports.scrollMargin = clsx_1.default;
//////////// Layout Utility functions
exports.display = clsx_1.default;
exports.boxDecorationBreak = clsx_1.default;
exports.boxSizing = clsx_1.default;
exports.container = clsx_1.default;
exports.float = clsx_1.default;
exports.clear = clsx_1.default;
exports.isolation = clsx_1.default;
exports.objectFit = clsx_1.default;
exports.overflow = clsx_1.default;
exports.overscrollBehavior = clsx_1.default;
exports.position = clsx_1.default;
exports.visibility = clsx_1.default;
exports.breakBefore = clsx_1.default;
exports.breakInside = clsx_1.default;
exports.breakAfter = clsx_1.default;
exports.objectPosition = clsx_1.default;
exports.inset = clsx_1.default;
exports.zIndex = clsx_1.default;
exports.aspectRatio = clsx_1.default;
exports.columns = clsx_1.default;
//////////// Sizing Utility functions
exports.width = clsx_1.default;
exports.minWidth = clsx_1.default;
exports.maxWidth = clsx_1.default;
exports.height = clsx_1.default;
exports.minHeight = clsx_1.default;
exports.maxHeight = clsx_1.default;
//////////// SVG Utility functions
exports.fill = clsx_1.default;
exports.stroke = clsx_1.default;
exports.strokeWidth = clsx_1.default;
//////////// Transforms Utility functions
exports.hardwareAcceleration = clsx_1.default;
exports.scale = clsx_1.default;
exports.rotate = clsx_1.default;
exports.translate = clsx_1.default;
exports.skew = clsx_1.default;
exports.transformOrigin = clsx_1.default;
//////////// Typography Utility functions
exports.fontSmoothing = clsx_1.default;
exports.fontStyle = clsx_1.default;
exports.fontVariantNumeric = clsx_1.default;
exports.listStylePosition = clsx_1.default;
exports.textAlign = clsx_1.default;
exports.textDecoration = clsx_1.default;
exports.textTransform = clsx_1.default;
exports.textOverflow = clsx_1.default;
exports.verticalAlign = clsx_1.default;
exports.whitespace = clsx_1.default;
exports.wordBreak = clsx_1.default;
exports.textDecorationStyle = clsx_1.default;
exports.fontFamily = clsx_1.default;
exports.fontSize = clsx_1.default;
exports.fontWeight = clsx_1.default;
exports.letterSpacing = clsx_1.default;
exports.lineHeight = clsx_1.default;
exports.listStyleType = clsx_1.default;
exports.placeholderColor = clsx_1.default;
exports.placeholderOpacity = clsx_1.default;
exports.textColor = clsx_1.default;
exports.textOpacity = clsx_1.default;
exports.content = clsx_1.default;
exports.textIndent = clsx_1.default;
exports.textDecorationColor = clsx_1.default;
exports.textDecorationThickness = clsx_1.default;
exports.textUnderlineOffset = clsx_1.default;
exports.TW = {
    screenReaders: exports.screenReaders,
    backgroundAttachment: exports.backgroundAttachment,
    backgroundClip: exports.backgroundClip,
    backgroundOrigin: exports.backgroundOrigin,
    backgroundRepeat: exports.backgroundRepeat,
    backgroundOpacity: exports.backgroundOpacity,
    backgroundColor: exports.backgroundColor,
    backgroundPosition: exports.backgroundPosition,
    backgroundSize: exports.backgroundSize,
    backgroundImage: exports.backgroundImage,
    gradientColorStops: exports.gradientColorStops,
    borderStyle: exports.borderStyle,
    divideStyle: exports.divideStyle,
    outlineStyle: exports.outlineStyle,
    borderColor: exports.borderColor,
    borderOpacity: exports.borderOpacity,
    borderRadius: exports.borderRadius,
    borderWidth: exports.borderWidth,
    divideColor: exports.divideColor,
    divideOpacity: exports.divideOpacity,
    divideWidth: exports.divideWidth,
    ringColor: exports.ringColor,
    ringWidth: exports.ringWidth,
    ringOpacity: exports.ringOpacity,
    ringOffsetColor: exports.ringOffsetColor,
    ringOffsetWidth: exports.ringOffsetWidth,
    outlineOffset: exports.outlineOffset,
    outlineWidth: exports.outlineWidth,
    outlineColor: exports.outlineColor,
    borderCollapse: exports.borderCollapse,
    tableLayout: exports.tableLayout,
    mixBlendMode: exports.mixBlendMode,
    backgroundBlendMode: exports.backgroundBlendMode,
    boxShadow: exports.boxShadow,
    boxShadowColor: exports.boxShadowColor,
    opacity: exports.opacity,
    transitionProperty: exports.transitionProperty,
    transitionDuration: exports.transitionDuration,
    transitionTimingFunction: exports.transitionTimingFunction,
    transitionDelay: exports.transitionDelay,
    animation: exports.animation,
    filter: exports.filter,
    backdropFilter: exports.backdropFilter,
    blur: exports.blur,
    brightness: exports.brightness,
    contrast: exports.contrast,
    dropShadow: exports.dropShadow,
    grayscale: exports.grayscale,
    hueRotate: exports.hueRotate,
    invert: exports.invert,
    saturate: exports.saturate,
    sepia: exports.sepia,
    backdropBlur: exports.backdropBlur,
    backdropBrightness: exports.backdropBrightness,
    backdropContrast: exports.backdropContrast,
    backdropGrayscale: exports.backdropGrayscale,
    backdropHueRotate: exports.backdropHueRotate,
    backdropInvert: exports.backdropInvert,
    backdropOpacity: exports.backdropOpacity,
    backdropSaturate: exports.backdropSaturate,
    backdropSepia: exports.backdropSepia,
    flexDirection: exports.flexDirection,
    flexWrap: exports.flexWrap,
    alignItems: exports.alignItems,
    alignContent: exports.alignContent,
    alignSelf: exports.alignSelf,
    placeContent: exports.placeContent,
    placeItems: exports.placeItems,
    placeSelf: exports.placeSelf,
    justifyContent: exports.justifyContent,
    justifyItems: exports.justifyItems,
    justifySelf: exports.justifySelf,
    flex: exports.flex,
    flexBasis: exports.flexBasis,
    flexGrow: exports.flexGrow,
    flexShrink: exports.flexShrink,
    order: exports.order,
    gridAutoFlow: exports.gridAutoFlow,
    gridTemplateColumns: exports.gridTemplateColumns,
    gridAutoColumns: exports.gridAutoColumns,
    gridColumn: exports.gridColumn,
    gridColumnStart: exports.gridColumnStart,
    gridColumnEnd: exports.gridColumnEnd,
    gridTemplateRows: exports.gridTemplateRows,
    gridAutoRows: exports.gridAutoRows,
    gridRow: exports.gridRow,
    gridRowStart: exports.gridRowStart,
    gridRowEnd: exports.gridRowEnd,
    gap: exports.gap,
    padding: exports.padding,
    margin: exports.margin,
    space: exports.space,
    appearance: exports.appearance,
    pointerEvents: exports.pointerEvents,
    resize: exports.resize,
    userSelect: exports.userSelect,
    scrollSnap: exports.scrollSnap,
    scrollBehavior: exports.scrollBehavior,
    touchAction: exports.touchAction,
    cursor: exports.cursor,
    caretColor: exports.caretColor,
    willChange: exports.willChange,
    accentColor: exports.accentColor,
    scrollPadding: exports.scrollPadding,
    scrollMargin: exports.scrollMargin,
    display: exports.display,
    boxDecorationBreak: exports.boxDecorationBreak,
    boxSizing: exports.boxSizing,
    container: exports.container,
    float: exports.float,
    clear: exports.clear,
    isolation: exports.isolation,
    objectFit: exports.objectFit,
    overflow: exports.overflow,
    overscrollBehavior: exports.overscrollBehavior,
    position: exports.position,
    visibility: exports.visibility,
    breakBefore: exports.breakBefore,
    breakInside: exports.breakInside,
    breakAfter: exports.breakAfter,
    objectPosition: exports.objectPosition,
    inset: exports.inset,
    zIndex: exports.zIndex,
    aspectRatio: exports.aspectRatio,
    columns: exports.columns,
    width: exports.width,
    minWidth: exports.minWidth,
    maxWidth: exports.maxWidth,
    height: exports.height,
    minHeight: exports.minHeight,
    maxHeight: exports.maxHeight,
    fill: exports.fill,
    stroke: exports.stroke,
    strokeWidth: exports.strokeWidth,
    hardwareAcceleration: exports.hardwareAcceleration,
    scale: exports.scale,
    rotate: exports.rotate,
    translate: exports.translate,
    skew: exports.skew,
    transformOrigin: exports.transformOrigin,
    fontSmoothing: exports.fontSmoothing,
    fontStyle: exports.fontStyle,
    fontVariantNumeric: exports.fontVariantNumeric,
    listStylePosition: exports.listStylePosition,
    textAlign: exports.textAlign,
    textDecoration: exports.textDecoration,
    textTransform: exports.textTransform,
    textOverflow: exports.textOverflow,
    verticalAlign: exports.verticalAlign,
    whitespace: exports.whitespace,
    wordBreak: exports.wordBreak,
    textDecorationStyle: exports.textDecorationStyle,
    fontFamily: exports.fontFamily,
    fontSize: exports.fontSize,
    fontWeight: exports.fontWeight,
    letterSpacing: exports.letterSpacing,
    lineHeight: exports.lineHeight,
    listStyleType: exports.listStyleType,
    placeholderColor: exports.placeholderColor,
    placeholderOpacity: exports.placeholderOpacity,
    textColor: exports.textColor,
    textOpacity: exports.textOpacity,
    content: exports.content,
    textIndent: exports.textIndent,
    textDecorationColor: exports.textDecorationColor,
    textDecorationThickness: exports.textDecorationThickness,
    textUnderlineOffset: exports.textUnderlineOffset,
};
exports.classnames = clsx_1.default;
exports.default = exports.classnames;
